apps/dashboard/app/controllers/scripts_controller.rb

```diff
@@ -40,6 +40,10 @@ def submit
 
     if (job_id = @script.submit(opts))
       redirect_to(project_path(params[:project_id]), notice: "Successfully submited job #{job_id}.")
+    elsif @script.cached_values
+      opts = @script.cached_values.to_h.symbolize_keys
+      job_id = @script.submit(opts)
+      redirect_to(project_path(params[:project_id]), notice: "Successfully submited job #{job_id}.")
```

Comment:
This could be causing some issues, and in any case - we don't need it here or to make `cached_values` public.

Really, it should just work in the view. You can see how recently used apps does this.  We already have the `smart_attributes` attr_reader and, they've been updated with the cached values in `build_smart_attributes`.


So in the view you can pull the `smart_attributes`

https://github.com/OSC/ondemand/blob/f9a02cfca01d54ef7b6f700060dc964f0eae7826/apps/dashboard/app/views/widgets/_recently_used_apps.html.erb#L10

Then just passes the parameters in the button itself.

https://github.com/OSC/ondemand/blob/f9a02cfca01d54ef7b6f700060dc964f0eae7826/apps/dashboard/app/views/widgets/_recently_used_apps.html.erb#L19